samples/nrf9160/modem_shell/src/main.c

```diff
@@ -257,7 +257,7 @@ void main(void)
 
 #if !defined(CONFIG_LWM2M_CARRIER)
 	/* Get Modem library initialization return value. */
-	err = modem_lib_init_result;
+	err = nrf_modem_lib_init(NORMAL_MODE);
```

Comment:
Is this change intentional?

Reply:
I probably need to improve this logic. We need to call `nrf_modem_lib_init` manually when modem tracing is enabled. Perhaps best to check if `CONFIG_NRF_MODEM_LIB_SYS_INIT` is disabled?

Reply:
Instead of checking `CONFIG_NRF_MODEM_LIB_SYS_INIT​` in main and have a code branch. How about I just move ​`CONFIG_NRF_MODEM_LIB_SYS_INIT=n`​ from `​overlay-modem-trace.conf`​ to `​prj.conf`​? So we always initialize the modem library manually (unless CONFIG_LWM2M_CARRIER).

Reply:
After thinking more about it, I think it might be unclear why `CONFIG_NRF_MODEM_LIB_SYS_INIT` would be disabled in prj.conf. So landed on keeping the configs as is and having a check in `main.c` instead.